driver/image_adjust_hue_driver.hpp

```diff
@@ -0,0 +1,389 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GUARD_MIOPEN_IMAGE_ADJUST_HUE_DRIVER_HPP
+#define GUARD_MIOPEN_IMAGE_ADJUST_HUE_DRIVER_HPP
+
+#include "../test/tensor_holder.hpp"
+#include "../test/verify.hpp"
+#include "InputFlags.hpp"
+#include "driver.hpp"
+#include "image_adjust_driver_common.hpp"
+#include "miopen/miopen.h"
+#include "miopen/tensor.hpp"
+#include "miopen/tensor_view.hpp"
+#include "random.hpp"
+#include "tensor_driver.hpp"
+#include "tensor_view.hpp"
+#include "timer.hpp"
+#include <cassert>
+#include <cmath>
+#include <memory>
+
+template <typename T = float>
+void mloConvertRGBToHSV(const T r, const T g, const T b, T* h, T* s, T* v)
+{
+    T minc = std::min(r, std::min(g, b));
+    T maxc = std::max(r, std::max(g, b));
+
+    *v = maxc;
+
+    T cr     = maxc - minc;
+    bool eqc = (cr == 0);
+
+    *s = cr / (eqc ? 1.0f : maxc);
+
+    T cr_divisor = eqc ? (T)1.0f : cr;
+    T rc         = (maxc - r) / cr_divisor;
+    T gc         = (maxc - g) / cr_divisor;
+    T bc         = (maxc - b) / cr_divisor;
+
+    T hr = (maxc == r) * (bc - gc);
+    T hg = ((maxc == g) && (maxc != r)) * ((T)2.0f + rc - bc);
+    T hb = ((maxc != g) && (maxc != r)) * ((T)4.0f + gc - rc);
+
+    *h = fmod((hr + hg + hb) / 6.0 + 1.0, 1.0);
+}
+
+template <typename T = float>
+void mloConvertHSVToRGB(const T h, const T s, const T v, T* r, T* g, T* b)
+{
+    T i        = floor(h * 6.0);
+    T f        = h * 6.0 - i;
+    int i_case = ((int)i + 6) % 6;
+
+    T p = clamp(v * (1.0 - s), 0.0, 1.0);
+    T q = clamp(v * (1.0 - s * f), 0.0, 1.0);
+    T t = clamp(v * (1.0 - s * (1.0 - f)), 0.0, 1.0);
+
+    switch(i_case)
+    {
+    case 0:
+        *r = v;
+        *g = t;
+        *b = p;
+        break;
+    case 1:
+        *r = q;
+        *g = v;
+        *b = p;
+        break;
+    case 2:
+        *r = p;
+        *g = v;
+        *b = t;
+        break;
+    case 3:
+        *r = p;
+        *g = q;
+        *b = v;
+        break;
+    case 4:
+        *r = t;
+        *g = p;
+        *b = v;
+        break;
+    case 5:
+        *r = v;
+        *g = p;
+        *b = q;
+        break;
+    default:
+        // Panic Get: How Did We Get Here?
```

Comment:
Please remove comment

Reply:
Changed the comment to make this part clear